### PR TITLE
add oneDNN graph backend for CPU

### PR DESCRIPTION
**Pull request by wenzhe-nrv**
This PR adds oneDNN graph backend to Dynamo. 
It is based on the oneDNN graph context manager API `fuser3` added here https://github.com/pytorch/pytorch/pull/82491

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: torchdynamo/optimizations/backends.py [hunk 1]
```diff
@@ -102,6 +102,12 @@ def nvfuser_ofi(subgraph):
         return reload_jit_model_ofi(subgraph)
 
 
+@create_backend
+def onednn_graph(subgraph):
```
Comment (jansel) [inline torchdynamo/optimizations/backends.py:106]:
```suggestion
def onednn(subgraph):
```
Using the string `"onednn"` seems nicer than `"onednn_graph"`

Comment (wenzhe-nrv) [reply torchdynamo/optimizations/backends.py:106]: Agree, that's nice without `_`. But `onednn_graph` is a bit different than onednn, thus using that might be miss leading. Let me think if there's a better naming.

Comment (wenzhe-nrv) [reply torchdynamo/optimizations/backends.py:106]: good suggest. I'll confirm the naming with our team. Thanks.

Comment (jansel) [reply torchdynamo/optimizations/backends.py:106]: The final call is up to you, happy to merge this name too.

Comment (wenzhe-nrv) [reply torchdynamo/optimizations/backends.py]: @jansel I updated the name. Thanks for suggesting.

**Pull request merged**